src/sync/index.js

```diff
@@ -58,316 +58,346 @@ function sequentialLoop(iterations, process, exit){
   return loop;
 }
 
-function sync(db){
+async function sync(db){
   const removeHexPrefix = true;
   var topicsNeedBalanceUpdate = new Set(), oraclesNeedBalanceUpdate = new Set();
 
-  qclient.getBlockCount()
-    .then(
-      (value)=>{
-        currentBlockChainHeight = value - 1;
-        var options = {
-          "limit": 1,
-          "sort": [["blockNum", 'desc']]
-        }
+  let currentBlockChainHeight = await qclient.getBlockCount();
+  currentBlockChainHeight = currentBlockChainHeight - 1;
+  let options = {
+    "limit": 1,
+    "sort": [["blockNum", 'desc']]
+  }
+
+  var startBlock = contractDeployedBlockNum;
+  let blocks = await db.Blocks.find({}, options).toArray();
+  if(blocks.length > 0){
+    startBlock = Math.max(blocks[0].blockNum + 1, startBlock);
+  }
+
+  var initialSync = sequentialLoop(Math.ceil((currentBlockChainHeight-startBlock)/batchSize), function(loop){
+    var endBlock = Math.min(startBlock + batchSize - 1, currentBlockChainHeight);
+    var syncTopic = false, syncCOracle = false, syncDOracle = false, syncVote = false, syncOracleResult = false, syncFinalResult = false;
+
+    // sync TopicCreated
+    var syncTopicCreatedPromise = new Promise(async (resolve) => {
+      let result;
+      try {
+        result = await qclient.searchLogs(startBlock, endBlock, Contracts.EventFactory.address, [Contracts.EventFactory.TopicCreated], Contracts, removeHexPrefix);
+        console.log('searchlog TopicCreated')
```

Comment:
is this missing a resolve() here?

Reply:
@dwalintukan nope. in the error-free flow, `resolve()` is called when db modification is doen